core/server/master/src/main/java/alluxio/master/file/DefaultFileSystemMaster.java

```diff
@@ -3402,23 +3406,26 @@ private boolean syncMetadata(RpcContext rpcContext, AlluxioURI path,
    *                                 of the permission checkers functions with the given inode path.
    *                                 If null, no permission checking is performed
    * @param isGetFileInfo            true if syncing for a getFileInfo operation
-   * @return
+   * @return syncStatus
    */
-  private boolean syncMetadata(RpcContext rpcContext, AlluxioURI path,
+  private InodeSyncStream.SyncStatus syncMetadata(RpcContext rpcContext, AlluxioURI path,
       FileSystemMasterCommonPOptions options, DescendantType syncDescendantType,
       @Nullable FileSystemMasterAuditContext auditContext,
       @Nullable Function<LockedInodePath, Inode> auditContextSrcInodeFunc,
       @Nullable PermissionCheckFunction permissionCheckOperation,
       boolean isGetFileInfo) throws AccessControlException, InvalidPathException {
     LockingScheme syncScheme = createSyncLockingScheme(path, options, isGetFileInfo);
-
     if (!syncScheme.shouldSync()) {
-      return false;
+      return NOT_NEEDED;
     }
```

Comment:
should we actually put following logic into `sync.sync()`?
```
    if (!syncScheme.shouldSync()) {
      return false;
      return NOT_NEEDED;
    }
```
and make `sync.sync` to return `InodeSyncStream.SyncStatus` instead of boolean?
i saw multiple different locations we call `InodeSyncStream#sync`, it will be good to fix it for all